### PR TITLE
INTLY-4182: Updating operator-sdk to v0.12.0

### PR DESCRIPTION
**Pull request by davidffrench**
## What
Updating the operator-sdk to v0.12.0. This upgrades `controller-runtime` and `controller-tools` from `0.1.12` to `0.3.0` which includes multiple breaking changes. Also migrating from dep to mod to manage dependencies.

### Breaking changes
1. Mod is unable to pull in enmasse resources from their GitHub repository.
2. client.List() parameters have switched position and opts require a new format
3. Log and Scheme dependencies have change path within the controller-runtime
4. Controller-utils createOrUpdate mutate function no longer takes a runtime object
5. CRD regenerated with new name and structural schema
6. log.logf.ZapLogger() no longer available

## Changes for each team member
1. Upgrade Go version to 1.13
2. Upgrade operator-sdk version to v0.12.0

## Verification Steps
1. E2E tests should pass

### CODE REVIEW

File: pkg/apis/enmasse/admin/v1beta1/register.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const GroupVersion = "v1beta1"
+
+var SchemeGroupVersion = schema.GroupVersion{Group: "enmasse.io", Version: GroupVersion}
```
Comment (matskiv) [inline pkg/apis/enmasse/admin/v1beta1/register.go]:
```suggestion
var SchemeGroupVersion = schema.GroupVersion{Group: "admin.enmasse.io", Version: GroupVersion}
```
Fixes this error:
```
"error":"product installation errors : failed installation of amqonline: 
could not create auth service &{{ } {standard-authservice  003-integreatly-amq-online ...
no matches for kind "AuthenticationService" in version "enmasse.io/v1beta1"
```

File: pkg/controller/installation/types.go [hunk 2]
```diff
@@ -92,12 +92,12 @@ var (
...
-				v1alpha1.ProductFuseOnOpenshift:        {Name: v1alpha1.ProductFuseOnOpenshift},
-				v1alpha1.ProductCodeReadyWorkspaces:    {Name: v1alpha1.ProductCodeReadyWorkspaces},
-				v1alpha1.ProductAMQOnline:              {Name: v1alpha1.ProductAMQOnline},
-				v1alpha1.Product3Scale:                 {Name: v1alpha1.Product3Scale},
+				v1alpha1.ProductLauncher:            {Name: v1alpha1.ProductLauncher},
+				v1alpha1.ProductFuse:                {Name: v1alpha1.ProductFuse},
+				v1alpha1.ProductFuseOnOpenshift:     {Name: v1alpha1.ProductFuseOnOpenshift},
+				v1alpha1.ProductCodeReadyWorkspaces: {Name: v1alpha1.ProductCodeReadyWorkspaces},
+				v1alpha1.ProductAMQOnline:           {Name: v1alpha1.ProductAMQOnline},
+				// v1alpha1.Product3Scale:                 {Name: v1alpha1.Product3Scale},
```
Comment (matskiv) [inline pkg/controller/installation/types.go]:
```suggestion
				v1alpha1.Product3Scale:                 {Name: v1alpha1.Product3Scale},
```

File: pkg/apis/enmasse/admin/v1beta1/register.go (hunk 1 above)
Comment (davidffrench) [reply pkg/apis/enmasse/admin/v1beta1/register.go]: Thanks @matskiv . Good catch

File: pkg/controller/installation/types.go (hunk 2 above)
Comment (davidffrench) [reply pkg/controller/installation/types.go]: Thanks @matskiv , good catch

**Comment by mikenairn** [comment]
/hold

Requires this change https://github.com/integr8ly/integreatly-operator/pull/211 and updates the the Dockerfile to install the correct versions of go and the operator-sdk

**Comment by matskiv** [comment]
Code looks good, I'll give it a spin on my cluster.

**Comment by matskiv** [comment]
Verified on my OSD cluster
/approve

**Pull request merged**